cpp/src/arrow/compute/function.h

```diff
@@ -235,6 +235,9 @@ class ARROW_EXPORT Function {
   virtual Result<Datum> Execute(const std::vector<Datum>& args,
                                 const FunctionOptions* options, ExecContext* ctx) const;
 
+  virtual Result<Datum> Execute(const ExecBatch& batch, const FunctionOptions* options,
+                                ExecContext* ctx) const;
```

Comment:
I suppose the plan is to expose a `ExecSpan`-taking version at some point?

Reply:
Unclear -- for those who care about performance, they instead want to deal with the kernel exec function directly. We can investigate when implementing a new-and-improved expression tree evaluator that doesn't have the bloat of the current implementation